### DISCUSSION TITLE
Problem with Security Level (?)

### DISCUSSION THREAD

**Discussion by nic410826** [comfyanonymous/ComfyUI#6264] [Q&A] [answered]
Hello.
I am pretty new to AI (this is my first thread on this site) and would appreciate to get some help from the community...  

I am using ComfyUI on my desktop (AMD Ryzen 7 7700X 8-Core Processor; NVIDIA GEForce RTX 4060, Windows 11 Pro). It happens that I cannot download certain models, custom nodes etc. which I would need for my workflows - apparently because of a security problem someware in my system.  In those cases I am getting the folloiwing error message: This action is not allowed with this security level configuration'. I tried various things (Deactivation / Deinstallation of Kaspersky, Windows Defender etc.) without any success. Then I installed ComfyUI on my Laptop (Windows 11 Pro) which connects via WLAN with my Network: On the Laptop I encountered none of the problems... 

Does anyone of you have / had the same problem and/or what would you suggest in order to solve the problem? Thank you very much for your feedback! Regards - Nicola  (nic410826)

![comfyui_manager_screenshot_002](https://github.com/user-attachments/assets/b2619336-09af-40d2-a751-48f6ad3cc6e4)

**Comment by LukeG89** [comment]
The “security level” that the error message is referring to is about the security policy of ComfyUI Manager: https://github.com/ltdrdata/ComfyUI-Manager?tab=readme-ov-file#security-policy

You can change the security level by editing the config.ini file in `ComfyUI\custom_nodes\ComfyUI-Manager`

To enable "Try fix" feature, you have to set the security level to `weak`

**Reply by nic410826** [reply]
Hi LukeG89. Thank you very much for your explanation and help! 
Regards nic410826

**Comment by ltdrdata** [comment]
Actually, the desktop version typically doesn't use **--listen**, so this shouldn't usually be an issue, but it might need to be checked.  
Could you upload the **comfyui.log** file?

**Reply by nic410826** [reply]
Hi ltdrdata. I am using 'ComfyUI_windows_portable'  on my desktop. Perhaps I wasn't precise enough... Does this change anything in your judgement? If not, how can I produce and upload the comfyui.log file?
Thanks and regards!

**Reply by ltdrdata** [reply]
Do you use `--listen` option? 

You can find the generated log path from terminal.

![image](https://github.com/user-attachments/assets/d80e3496-3a16-40dd-90a9-845775338b2e)

**Reply by nic410826** [reply]
Hello ltdrdata.
I am sending you the comfyui-Logs of my PC (250101_comfyui_pc) and of my Laptop (250101_comfyui_laptop), on which I don't have the problem with the security. By the way, the suggestion of a community member to change the settings in the init-File from 'normal' to 'weak' didn't work. The problem with the security level om my PC does still exist and is hindering me setting up further workflows in Flux and Stable Diffusion 3.5... In the meantime I had further discussions with support people of my security software and of my PC. The 'worst' proposal is the complete reinstallation of the software on my PC which I wouldnt like to do before I know the reason for the problem. In order to show you the consequences of the problem I have als enclosed a screen shot of the custom nodes which the ComfyUI Manager isn't able to load. I really appreciate your help and I am looking foreward for any good news...
Thank you ver much and best regards - NICOLA

[250101_comfyui_pc.log](https://github.com/user-attachments/files/18284898/250101_comfyui_pc.log)
[250101_comfyui_laptop.log](https://github.com/user-attachments/files/18284902/250101_comfyui_laptop.log)
[250101_custom-nodes-with-import-fails.docx](https://github.com/user-attachments/files/18285015/250101_custom-nodes-with-import-fails.docx)

**Reply by nic410826** [reply]
... and an addition: I am NOT using the --listen option

**Reply by nic410826** [reply]
Hello ltdrdata.
My strange 'security problem' seems to be solved. LukeG89 proposed to change the path-name (too long and blanks in it) of my ComfyUI_windows-portable. I did it and was able to download all the coustom nodes with the download problem except only three (#92, #132 and #137)I am very happy that the proposal of LukeG39 has solved the problem! And I would like to thank you as well for your support and assistance. Best regards - Nicola

**Comment by nic410826** [comment]


**Reply by LukeG89** [reply]
I think that the problem in your pc is caused by the folder path:
`C:\Users\NICOLA\Desktop\ComfyUI - Rüst\ComfyUI_windows_portable\ComfyUI`

ComfyUI doesn't like:
- long paths
- spaces in the path

I would try:
`C:\Users\NICOLA\ComfyUI_windows_portable\ComfyUI`

Or maybe just correcting `\ComfyUI - Rüst\` would fix the issue

**Reply by nic410826** [reply]
Hi LukeG89.
Thank you for your reply! I will make the suggested correction and hope this will solve the problem...
Regards - NICOLA

**Reply by nic410826** [reply, marked as answer]
Hi LukeG89.
I have very good news: I modified the path of ComfyUI... as you proposed and it seems that this has solved my strange 'security problem'. There are only 3 custom nodes (# 92, #132 and #137) which I couldn't still not download after the path-modification; but I assume that this 'problem' has other reasons... 
I would like to thank you very much for your great help! I am very happy that everything is working fine by now.
Again, many thanks and best regards - Nicola

**Reply by LukeG89** [reply]
Great! 🎉 
Maybe the custom nodes that still can't be imported are missing a dependency (some python packages are not easy to install, like insightface for example) or there is a dependency conflict.

Anyways, I noticed that you have installed two problematic custom nodes:
- ComfyUI-Flow-Control
- comfyui-tensorops

These two nodes are just creating conflicts with other nodes, you probably install them by mistake after importing a workflow with ComfyUI-GGUF and/or ComfyUI-Florence2 missing nodes (Check [this](https://github.com/ltdrdata/ComfyUI-Manager/issues/1362) and [this](https://github.com/ltdrdata/ComfyUI-Manager/issues/1378#issuecomment-2564666698)). You should uninstall them.

**Reply by offmybach** [reply]
Got rid of Flow control and tensorops. changed security to weak. short log path with no spaces. still running into without permission due to security in portable version.      [default]
preview_method = none
badge_mode = none
git_exe = 
channel_url = https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main
share_option = all
bypass_ssl = False
file_logging = True
default_ui = none
component_policy = workflow
double_click_policy = copy-all
windows_selector_event_loop_policy = False
model_download_by_agent = False
downgrade_blacklist = 
security_level = weak

**Reply by LukeG89** [reply]
@offmybach If your Manager is up-to-date to the latest, now it uses another config.ini file located in `ComfyUI\user\default\ComfyUI-Manager`, maybe this is the problem for you

**Comment by offmybach** [comment]
that was it! thanks so much for your help

On Wed, Jan 15, 2025 at 10:31 AM LukeG89 ***@***.***> wrote:

> @offmybach <https://github.com/offmybach> If your Manager is up-to-date
> to the latest, now it uses another config.ini file located in
> ComfyUI\user\default\ComfyUI-Manager, maybe this is the problem for you
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/comfyanonymous/ComfyUI/discussions/6264#discussioncomment-11844464>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BA3LYDEDSK6QKW2WWO44LV32KZ5NHAVCNFSM6AAAAABUK6OG4KVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOBUGQ2DMNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


--
